# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/XPCJSRuntime.cpp
# Commit: 71d2bf08d2da
# Full Hash: 71d2bf08d2da8edd6e390cf38eee723d8d4733d8
# Author: Denis Palmeiro <dpalmeiro@mozilla.com>
# Date: 2025-01-21 21:28:51
# Description:
#   Bug 1942415: Revert changes for GC_GLEAN_SLOW_PHASE and GC_GLEAN_SLOW_TASK probes.  r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234808
# ==============================================================================

diff -r 19e1106de68c -r 71d2bf08d2da js/xpconnect/src/XPCJSRuntime.cpp
--- a/js/xpconnect/src/XPCJSRuntime.cpp	Tue Jan 21 16:08:01 2025 +0200
+++ b/js/xpconnect/src/XPCJSRuntime.cpp	Tue Jan 21 14:28:44 2025 +0000
@@ -2814,20 +2814,6 @@
           JS::ExplainGCReason(static_cast<JS::GCReason>(sample)));
       glean::javascript_gc::reason.Get(reason).Add(1);
     } break;
-    case JSMetric::GC_GLEAN_SLOW_PHASE: {
-      MOZ_ASSERT(sample < static_cast<uint32_t>(
-                              glean::javascript_gc::SlowPhaseLabel::e__Other__),
-                 "Phase does not exist in the slow_phase labels list.");
-      nsAutoCString phase(JS::GetGCPhaseName(sample));
-      glean::javascript_gc::slow_phase.Get(phase).Add(1);
-    } break;
-    case JSMetric::GC_GLEAN_SLOW_TASK: {
-      MOZ_ASSERT(sample < static_cast<uint32_t>(
-                              glean::javascript_gc::SlowTaskLabel::e__Other__),
-                 "Phase does not exist in the slow_task labels list.");
-      nsAutoCString phase(JS::GetGCPhaseName(sample));
-      glean::javascript_gc::slow_task.Get(phase).Add(1);
-    } break;
     case JSMetric::GC_RESET_REASON: {
       MOZ_ASSERT(
           sample < static_cast<uint32_t>(
